File: my-credentials-plugin/src/main/java/holygradle/credentials/CancelTimerMouseListener.java
package holygradle.credentials;

import java.awt.event.*;
import java.util.*;
 
class CancelTimerMouseListener implements MouseListener {
    private final Timer timer;

    public CancelTimerMouseListener(Timer timer) {
        this.timer = timer;
    }
    
    public void	mouseClicked(MouseEvent e) {
        timer.cancel();
    }
 
    public void mouseEntered(MouseEvent e) {
    }
    
    public void mouseExited(MouseEvent e) {
    }
    
    public void mousePressed(MouseEvent e) {
    }
    
    public void mouseReleased(MouseEvent e) {
    }
}

File: my-credentials-plugin/src/main/kotlin/holygradle/credentials/CancelTimerMouseListener.kt
package holygradle.credentials

import java.awt.event.*
import java.util.*
 
class CancelTimerMouseListener(val timer: Timer) : MouseListener {
    override fun mouseClicked(e: MouseEvent) {
        timer.cancel()
    }

    override fun mouseReleased(e: MouseEvent?) {
    }

    override fun mouseEntered(e: MouseEvent?) {
    }

    override fun mouseExited(e: MouseEvent?) {
    }

    override fun mousePressed(e: MouseEvent?) {
    }
}